keystore/src/connection/mod.rs

```diff
@@ -192,8 +194,31 @@ impl Connection {
         Ok(())
     }
 
-    pub async fn new_transaction(&self) -> CryptoKeystoreResult<KeystoreTransaction> {
-        KeystoreTransaction::new(self.clone()).await
+    pub async fn new_transaction(&self) -> CryptoKeystoreResult<()> {
+        let mut transaction = self.transaction.lock().await;
+        if transaction.is_some() {
+            return Err(CryptoKeystoreError::TransactionInProgress);
+        }
+        *transaction = Some(KeystoreTransaction::new(self.clone()).await?);
+        Ok(())
+    }
+
+    pub async fn child_groups<
+        E: Entity<ConnectionType = KeystoreDatabaseConnection>
+        + crate::entities::PersistedMlsGroupExt
+        + Sync,
+    >(
+        &self,
+        entity: E,
+    ) -> CryptoKeystoreResult<Vec<E>> {
+        let mut conn = self.conn.lock().await;
+        let persisted_records = entity.child_groups(conn.deref_mut()).await?;
+
+        let transaction_guard = self.transaction.lock().await;
+        let Some(transaction) = transaction_guard.as_ref() else {
+            return Ok(persisted_records);
+        };
+        transaction.child_groups(entity, persisted_records).await
     }
```

Comment:
I don't understand this function, could you please described what it is supposed to do?

Reply:
It collects all records that have the  id of `entity` as `parent_id`. I don't know in what context this may be needed, AFAICT this function isn't used anywhere. Should we remove it in this PR or in another?